### PR TITLE
[RFC]: metrics: report: Add debug, symbolise scaling node data extraction

### PR DESCRIPTION
**Pull request by grahamwhaley**
Two semi related commits (yeah, really I should have split to two pull requests....)

- When using the makereport script in debug mode, map the scripts dir in from the host so you can 'live edit' the scripts from the host and test them in the container (and, yes, I should add that to the documentation).
- Update the node_util data parsing to use symbolic lookups rather than using array position knowledge, to detach the code from knowing the physical layout of the data - this will allow us to add more data into the structure and have the parsing still work, as long as we don't change the data symbolic names.

### CODE REVIEW

**Comment by grahamwhaley** [comment]
@dklyle - an RFC, as I'd like your eyes on this (as I am modifying your code), and I should also update the README (which I'll do tomorrow).
The debug item is useful though ;-)

File: metrics/report/report_dockerfile/scaling.R [hunk 1]
```diff
@@ -59,43 +59,32 @@ for (currentdir in resultdirs) {
...
+			# Gather up each 'column' from the dataframe - one per run instance.
+			# The length(node) parts are as each entry has an n-node array of
+			# results, so each iteration can create more than one new row in the
+			# final table.
+			fudata=c()
+			furoot=fdata$BootResults
+			for (n in seq(length(furoot$n_pods$Result))) {
+				num_pods=furoot$n_pods$Result[[n]]
+				u=furoot$node_util[[n]]
+				# first iteration provide column name for c1
```
Comment (grahamwhaley) [inline metrics/report/report_dockerfile/scaling.R]: Hmm, maybe I need to nuke that comment line, and/or add back in some column naming (which can be done at the end - assigning a c(names) to colnames(fudata) iirc).

File: metrics/report/report_dockerfile/scaling.R [hunk 2]
```diff
@@ -59,43 +59,32 @@ for (currentdir in resultdirs) {
...
+				c1=cbind(node=u$node)
+				c2=cbind(noschedule=u$noschedule)
+				c3=cbind(cpu_idle=u$cpu_idle$Result)
+				c4=cbind(mem_free=u$mem_free$Result)/(1024*1024)
+				# using index to make chart start with 0 rather than 1
```
Comment (grahamwhaley) [inline metrics/report/report_dockerfile/scaling.R]: and.... @dklyle - let me know if I've messed up the 0 indexing... the graph I think looked OK (both old and new versions indexed from 0?). I'll check again though. heh, knew this should all be an RFC ;-)

Comment (dklyle) [reply metrics/report/report_dockerfile/scaling.R]: I did not get to try this today, but it looks reasonable to me. That comment should have been removed with my last fix. Feel free to remove it.

File: metrics/report/report_dockerfile/scaling.R (hunk 1 above)
Comment (dklyle) [reply metrics/report/report_dockerfile/scaling.R]: yes, can kill this comment

**Comment by dklyle** [comment]
This looks good to me. I'll approve when you say you're ready.

**Comment by grahamwhaley** [comment]
I'm going to close/deprecate this one - it inspired me to do further updates - new PR imminent...

**Pull request closed without merging**